### PR TITLE
Fix bugs found in Publisher Subscription config page

### PR DESCRIPTION
**Pull request by Arshardh**
Fixes https://github.com/wso2/product-apim/issues/6502

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Subscriptions/Subscriptions.jsx [hunk 1]
```diff
@@ -53,8 +57,41 @@ function Subscriptions(props) {
...
+            })
+            .catch((error) => {
+                console.error(error);
+                Alert.error('Error occurred while updating subscription configurations');
+            }).finally(() => {
+                setUpdateInProgress(false);
+            });
+    }
+
+
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Subscriptions/Subscriptions.jsx]: Shall we remove the extra line?

**Comment by Arshardh** [comment]
image

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Subscriptions/Subscriptions.jsx (hunk 1 above)
Comment (Arshardh) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Subscriptions/Subscriptions.jsx]: done

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Subscriptions/Subscriptions.jsx [hunk 2]
```diff
@@ -53,8 +57,40 @@ function Subscriptions(props) {
...
+                console.error(error);
+                Alert.error('Error occurred while updating subscription configurations');
+            }).finally(() => {
+                setUpdateInProgress(false);
+            });
+    }
+
     useEffect(() => {
+        if (api) {
+            setAvailability({ subscriptionAvailability: api.subscriptionAvailability });
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Subscriptions/Subscriptions.jsx]: why we set this inside an effect??

Comment (Arshardh) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Subscriptions/Subscriptions.jsx]: I want to set it once the component is loaded. can I set this when I am defining the state variable instead?

**Pull request merged**